### ISSUE TITLE
Add obstacle avoidance function to the urban example

### ISSUE THREAD

**Issue by ncnynl** [open-navigation/opennav_amd_demonstrations#54]
Hello, we have some questions when using this program. In the urban_navigation_demo.py example, this can only be processed according to the planned path. Can the dynamic obstacle avoidance function be added?

**Comment by SteveMacenski** [comment]
Certainly, but that isn't part of this POC demo (and not necessary for everyone's needs for that type of navigation).

**Comment by ncnynl** [comment]
Thanks for reply

I am quite interested in this, and I have written a special Chinese column to introduce this repo on our website `https://www.ncnynl.com/category/ros2-opennav/`

We have configured an installation script to quickly deploy this code `https://gitee.com/ncnynl/commands/blob/master/commands_extra/ros2_gazebo/shell/install_ros2_opennav_ws.sh`

I want to expand the application more, such as adding rtabmap, Orb-slam, and other visual mapping and navigation. Then expand more multi-line laser mapping algorithms, so that it becomes a more complete application demo

There is still a question about the navigation example of ubran. Is it just using line following without really using 3D positioning? I haven't figured this part out yet.
